src/attention/utils/helpers.ts

```diff
@@ -153,26 +162,58 @@ export async function useRecompute(state: AttentionState) {
       });
     }
 
+    // Handle visibility based on hide middleware
     if (middlewareData?.hide && !state?.isCallout) {
       const { referenceHidden } = middlewareData.hide;
       Object.assign(attentionEl.style, {
         visibility: referenceHidden ? 'hidden' : '',
       });
     }
 
+    // Arrow position adjustment
     if (middlewareData?.arrow && state?.arrowEl) {
       const arrowEl: HTMLElement = state?.arrowEl as HTMLElement;
       const { x: arrowX, y: arrowY } = middlewareData.arrow;
-      const { x: shiftX } = middlewareData.shift || {}; // Get the shift value if it exists
-
-      // Adjust the arrow based on the cross-axis shift
-      let left = typeof arrowX === 'number' ? `${arrowX - (shiftX || 0)}px` : '';
-      let top = typeof arrowY === 'number' ? `${arrowY}px` : '';
-
+      const isRtl = window.getComputedStyle(attentionEl).direction === 'rtl'; // Checks RTL for proper arrow alignment
+      const arrowPlacement: string = arrowDirection(placement).split('-')[1];
+
+      let top = '',
+        right = '',
+        bottom = '',
+        left = '';
+
+      // Adjust based on 'start' or 'end' placements
+      if (arrowPlacement === 'start') {
+        const value = typeof arrowX === 'number' ? `calc(24px - ${arrowEl.offsetWidth / 2}px)` : '';
```

Comment:
Is the `24px` a value that is in any way dependant on the `8px` padding you added in this PR? I'm asking in case either of the two need to change. If they are related, maybe we should assign these values to some constant or in some other way make it easy to know what should be updated?

Reply:
The `24px` value used in the arrow positioning calculation is not directly related to the `8px` padding added to the entire `attentionEl`. The `24px` is used to control the arrow's alignment, ensuring it’s correctly positioned relative to its container (often aligning it with the middle or adjusting based on the placement, e.g., `"start"` or `"end"`). While the `8px` padding is applied to the content of the `attentionEl` to prevent it from overflowing the viewport.

However,  I agree that both values could be moved into constants, which would help with improving readability and maintainability, should we want to change these values in the future. :) I could create something like this: 
```
const ARROW_OFFSET = 24;
const ELEMENT_PADDING = 8;
```

Reply:
Updated in this commit: [9c256f8](https://github.com/warp-ds/core/pull/22/commits/9c256f8fce135824656222ccc22933af7378c890)